src/github_repo_request_local.py

```diff
@@ -83,45 +165,152 @@ def list_test_files(directory, excluded_extensions):
+        if row["testfilecountlocal"] != -1 and pd.notna(row["last_commit_hash"]):
+            continue
+
+        original_repo_url = row["repourl"]
+        repo_url = get_base_repo_url(original_repo_url)
+        if not repo_url or repo_url is None:
+            logger.info(f"Invalid repository URL: {original_repo_url}")
             continue
+        repo_name = Path(repo_url.split("/")[-1]).stem
+        clone_dir = clone_dir_base / repo_name
+
+        # Clone only if directory doesn't exist
+        if not clone_dir.exists():
+            try:
+                logger.info(f"Trying to clone {repo_url} into {clone_dir}")
+                subprocess.run(
+                    ["git", "clone", repo_url, str(clone_dir)],
+                    check=True,
+                    capture_output=True,
+                )
+                logger.info(f"Successfully cloned the repo: {repo_name}")
+
+                # Fetch the last commit hash and store it in the DataFrame
+                last_commit_hash = get_last_commit_hash(clone_dir)
+                if last_commit_hash is not None:
+                    df.at[index, "last_commit_hash"] = last_commit_hash
+
+                # On successful clone, increment the processed_count
+                processed_count += 1
+
+            except subprocess.CalledProcessError as e:
+                logger.error(
+                    f"Failed to clone the repo: {repo_name}." f" " f"Exception: {e}"
+                )
+                df.at[index, "testfilecountlocal"] = -1
+
+                # Even on failure, consider it processed for this batch
+                processed_count += 1
+
+                continue
+
+        # Always attempt to fetch the last commit hash if not already fetched
+        if pd.isna(row["last_commit_hash"]):
+            last_commit_hash = get_last_commit_hash(clone_dir)
+            df.at[index, "last_commit_hash"] = last_commit_hash
 
-    test_file_names = list_test_files(clone_dir, args.exclude)
-    count = len(test_file_names)
-    df.at[index, "testfilecountlocal"] = count
-    formatted_names = "\n".join(test_file_names)
-    logger.info(f"Test file names for {repo_name}:\n{formatted_names}")
+        # Count test files if not already counted
+        if row["testfilecountlocal"] == -1:
+            test_file_names = list_test_files(clone_dir, args.exclude)
+            count = len(test_file_names)
+            df.at[index, "testfilecountlocal"] = count
 
-    # Save after each update
+            # Write the repository URL and each test file name to the text file
+            file.write(f"Repository URL: {repo_url}\n")  # Write the repo URL
+            file.writelines(
+                f"{name}\n" for name in test_file_names
+            )  # Write each test file name
+            file.write("\n")  # Add a blank line for separation
+
+            formatted_names = "\n".join(test_file_names)
+            logger.info(f"Test file names for {repo_name}:\n{formatted_names}")
```

Comment:
I get the impression we're writing the names of the tests twice now, once to the log (here) and once to a separate file. I don't think we need to write these details to the log otherwise the log file fills rapidly and it's hard to see what else is happening during the execution of the script. Write summary info to the log, details to the separate file.

Reply:
- Removed console output of test file names during repository processing